### ISSUE TITLE
OSError: Too many open files

### ISSUE THREAD

**Issue by 23doors** [celery/celery#887]
Just tried upgrading celery from 2.4.6 to 3.0.4 and kombu from 2.1.0 to 2.3.0.

Got same issue just like here:
https://groups.google.com/forum/#!msg/celery-users/_MWg_oD_NRQ/H4CyG-YClDEJ
Tried with EXECV=True, no go.

OS: OSX Snow Leopard

2.4.6 works flawlessly.

**Comment by gcmartinelli** [comment]
Is there a solution for this? I'm facing the same issue. 
I'm using Celery 4.3.0 and Billiard, Supervisor 4.0.4 and RabbitMQ on Ubuntu 18.4.

My task is running on Beats and involves socket connections. I get about 30-1h of no issues but after that some workers start failing with the error msg:
ERROR/ForkPoolWorker-2] Too many open files.

I have increased the limits on my the open files of my system, it just extended the time until failure. My task is running inside a control-flow ("with" statement) which should be closing pools upon completion.

**Comment by thedrow** [comment]
This may be related to a different issue in billiard.
See https://github.com/celery/billiard/issues/282.

**Comment by gcmartinelli** [comment]
I couldn't pinpoint the issue... but so far I've solved my problems by taking these steps:
1) Dropped RabbitMQ for Redis
2) Refactored my code, dropping the usage of `billiard`. Instead I broke my task in two (one periodic task and one job) and started using `celery.group` to run these jobs in parallel.
3) I was instantiating celery with `celery -A app worker -B` (plus some other options). Instead, I broke into two processes, one worker (`celery -A app worker`) and one beat (`celery -A app beat`). Both run by `supervisor`.
